core/include/usFramework.h

```diff
@@ -0,0 +1,155 @@
+{
+public:
+    virtual ~Framework(void);
+
+    /**
+     * Initialize this Framework.
+     *
+     * @throw std::runtime_error If this Framework could not be initialized.
+     */
+    void init(void);
+
+    /**
+     * Start this Framework.
+     *
+     * <p>
+     * The following steps are taken to start this Framework:
+     * -# If this Framework is not in the STARTING state, initialize this Framework.
+     * -# This Framework's state is set to ACTIVE.
+     *
+     * @throws std::runtime_error If this Framework could not be started.
+     */
+    void Start();
+
+    /**
+     * Stop this Framework.
+     *
+     * The following steps are taken to stop this Framework:
+     * -# This Framework's state is set to STOPPING.
+     * -# All installed bundles must be stopped.
+     * -# Unregister all services registered by this Framework.
+     * -# Event handling is disabled.
+     * -# All resources held by this Framework are released. This includes threads, open files, etc.
+     *
+     * After being stopped, this Framework may be discarded, initialized or started.
+     *
+     * @throws std::runtime_error If stopping this Framework could not be initiated.
+     */
+    void Stop();
+
+    /**
+     * The Framework cannot be uninstalled.
+     *
+     * This method always throws a std::runtime_error exception.
+     * 
+     * @throws std::runtime_error This Framework cannot be uninstalled.
+     */
+    void Uninstall();
+
+    /**
+    * Returns this Framework's location.
+    *
+    * <p>
+    * This Framework is assigned the unique location "System Bundle" 
+    * since this Framework is also a System Bundle.
+    *
+    * @return The string "System Bundle".
+    */
+    std::string GetLocation() const;
+
+    /**
+     * Enable or disable auto-install support.
+     *
+     * @param enable If \c true, enable auto-install support, disable it otherwise.
+     *
+     * @remarks Calling this method will have no effect if support for
+     * auto-loading has not been configured into the CppMicroServices library of it
+     * it has been disabled by defining the US_DISABLE_AUTOLOADING envrionment variable.
+     *
+     * @deprecated This method remains for legacy clients and will be removed in a 
+     * future release. Auto-loading support will be moved into a seprate service,
+     * at which point, this method will be permanently removed.
+     * 
+     */
+    void SetAutoLoadingEnabled(bool enable);
+
+private:
+    // Framework instances are exclusively constructed by the FrameworkFactory class
+    friend class FrameworkFactory;
+    
+    // Allow the framework to be constructed with configuration properties
+    // provided by a FrameworkFactory object.
+    Framework(std::map<std::string, std::string>& configuration);
+    Framework(void);
+
+    // This class is not copy-able
+    Framework(const Framework&);
+    Framework operator=(const Framework&);
+
+    FrameworkPrivate* f;
```

Comment:
Naming conventions for private impl pointers is `d`. Please use `FrameworkPrivate* d`.